File: app/src/main/java/rod/bailey/trafficatnsw/json/hazard/XPeriod.java
package rod.bailey.trafficatnsw.json.hazard;

import org.json.JSONObject;

import static rod.bailey.trafficatnsw.util.JSONUtils.*;

public class XPeriod {

	private String closureType;
	private String direction;
	private String finishTime;
	private String fromDay;
	private String startTime;
	private String toDay;

	public XPeriod(JSONObject json) {
		assert json != null;

		closureType = safeGetString(json, "closureType");
		direction = safeGetString(json, "direction");
		finishTime = safeGetString(json, "finishTime");
		fromDay = safeGetString(json, "fromDay");
		startTime = safeGetString(json, "startTime");
		toDay = safeGetString(json, "toDay");
	}

	public String getClosureType() {
		return closureType;
	}

	public String getDirection() {
		return direction;
	}

	public String getFinishTime() {
		return finishTime;
	}

	public String getFromDay() {
		return fromDay;
	}

	public String getStartTime() {
		return startTime;
	}

	public String getToDay() {
		return toDay;
	}

}

File: app/src/main/java/rod/bailey/trafficatnsw/json/hazard/XPeriod.kt
package rod.bailey.trafficatnsw.json.hazard

import org.json.JSONObject

import rod.bailey.trafficatnsw.util.JSONUtils.*

class XPeriod(json: JSONObject) {

	val closureType: String?
	val direction: String?
	val finishTime: String?
	val fromDay: String?
	val startTime: String?
	val toDay: String?

	init {
		closureType = safeGetString(json, "closureType")
		direction = safeGetString(json, "direction")
		finishTime = safeGetString(json, "finishTime")
		fromDay = safeGetString(json, "fromDay")
		startTime = safeGetString(json, "startTime")
		toDay = safeGetString(json, "toDay")
	}

}
